can/tests/test_dispatch.cpp

```diff
@@ -164,4 +164,43 @@ SCENARIO("DispatchParseTarget") {
             }
         }
     }
+}
+
+SCENARIO("DispatchParseTargetNode") {
+    using BufferType = std::array<uint8_t, 0>;
+    using MessageTypes =
+        std::variant<std::monostate, HeartbeatRequest, HeartbeatResponse>;
+
+    struct Handler {
+        Handler() {}
+        Handler(const Handler&) = delete;
+        Handler(const Handler&&) = delete;
+        Handler& operator=(const Handler&) = delete;
+        Handler&& operator=(const Handler&&) = delete;
+
+        void handle(MessageTypes& m) { parsed_message = m; }
+        MessageTypes parsed_message{};
+    };
+
+    GIVEN(
+        "A DispatchParseTargetNode that accepts hearbeat request and response "
+        "types for head's left motor") {
+        auto l = Handler{};
+        auto buff = BufferType{};
+        auto node = 0x60;
+        auto subject = DispatchParseTargetNode<Handler, HeartbeatRequest,
+                                               HeartbeatResponse>(l, node);
+
+        WHEN("Given a HeartbeatRequest") {
```

Comment:
let's also add a test that if the node id part is different but the heartbeat message id part is correct, this parser does _not_ call the handler